File: qc-core/src/main/java/com/eager/questioncloud/core/domain/post/infrastructure/repository/PostCommentRepository.java
package com.eager.questioncloud.core.domain.post.infrastructure.repository;

import com.eager.questioncloud.core.common.PagingInformation;
import com.eager.questioncloud.core.domain.post.dto.PostCommentDetail;
import com.eager.questioncloud.core.domain.post.model.PostComment;
import java.util.List;

public interface PostCommentRepository {
    PostComment save(PostComment postComment);

    PostComment findByIdAndWriterId(Long commentId, Long userId);

    List<PostCommentDetail> getPostCommentDetails(Long postId, Long userId, PagingInformation pagingInformation);

    void delete(PostComment postComment);

    int count(Long postId);
}

File: qc-core/src/main/kotlin/com/eager/questioncloud/core/domain/post/infrastructure/repository/PostCommentRepository.kt
package com.eager.questioncloud.core.domain.post.infrastructure.repository

import com.eager.questioncloud.core.common.PagingInformation
import com.eager.questioncloud.core.domain.post.model.PostComment

interface PostCommentRepository {
    fun save(postComment: PostComment): PostComment

    fun findByIdAndWriterId(commentId: Long, userId: Long): PostComment

    fun findByPostIdWithPagination(
        postId: Long,
        pagingInformation: PagingInformation
    ): List<PostComment>

    fun delete(postComment: PostComment)

    fun count(postId: Long): Int
}
